build_tools/azure/py38_conda_defaults_openblas_environment.yml

```diff
@@ -0,0 +1,20 @@
+channels:
+  - defaults
+dependencies:
+  - python=3.8
+  - numpy=1.17.3
```

Comment:
would it be possible to have a way to use a specific syntax such as:

```
    - numpy=MIN_SUPPORTED
```

that would be preprocessed to be replaced by:

```
    - numpy=1.17.3
```

by using `sklearn/_min_dependencies.py` before calling conda-lock on the result of the replacement?

Reply:
I kind of agree that we lose the min (or latest) version thing and also the ability to have all versions defined in the same place (what we partially had in `azure-pipelines.yml` before although it was also in `build_tools/azure/install.sh` for some things).

What I had in mind: a script with all the versions that would generate the `environment.yml` for all the builds. Maybe not in this PR though?

Reply:
> I kind of agree that we lose the min (or latest) version

The latest is not lost. If the version is not defined in the environment.yml template file, then it's assumed "latest", provided that the future conda-lock update bot can issue and merge a non-failing upgrade PR to effectively track the actual latest version.

> What I had in mind: a script with all the package + versions for all the builds that would generate the environment.yml for all the builds. Maybe not in this PR though?

I would rather like to have a synchronization script for to handle the case of the `"min"` specs in the CI as part of this PR personally.